packages/api-specification/api/windows.ts

```diff
@@ -0,0 +1,313 @@
+import {Bounds, UnsubscribeFunction} from "./common";
+import {Base64ImageData, CaptureOptions} from "./system";
+import EventEmitter = NodeJS.EventEmitter;
+
+export interface WindowsAPI {
+
+    /** Returns the current window */
+    current: Window;
+
+    /** Returns all windows in the application */
+    all: Window[];
+
+    /**
+     * The window that is focused in this application, otherwise returns null
+     */
+    focusedWindow: Window;
+
+    /**
+     getById(windowId: string): Window;
+
+     /** Opens a new window */
+    open(options: WindowOptions): Promise<Window>;
+
+    /**
+     * Receive notifications when a new window is opened
+     * @returns Function Execute the function to unsubscribe
+     */
+    onWindowOpened(callback: (window: Window) => void): UnsubscribeFunction;

```

Comment:
Is there any reason why `Window` is an event emitter, whereas the `WindowsAPI` has explicit methods for each event type?

I'd prefer to stick to the event emitter pattern!